### DISCUSSION TITLE
Frame data

### DISCUSSION THREAD

**Discussion by TruXail** [ikemen-engine/Ikemen-GO#1603] [Feature Request] [answered]
### Is your feature request related to a problem? Please describe.

There is currently no way to look at how many frames a character's attack comes out in without going into fighter factory and having to manually look up the move. In characters with particularly large sheets, this becomes an extremely long and tedious process.

### Describe the solution you'd like

A system that calculates how many frames an attack or special takes to come out or "activate" from the beginning of the button press. Additionally a system that calculates Hit stun frame data via checking how many frames of recovery the user has in comparison to the opponent on hit or block.

This would additionally be useful due to there being many differences between characters and it often being ambiguous as to what combos\block strings work between different versions of the same character or characters drastically changed from their origional counterpart.

### Additional context

Here as an example using GBVSR:

![GBVSR-Win64-Shipping_tofFmCJDj4](https://github.com/ikemen-engine/Ikemen-GO/assets/97189521/e6a8a335-3992-4862-b242-1622659da06d)

**Comment by potsmugen** [comment, marked as answer]
This sounds a bit too elaborate for the included training mode, I think. You can download modules that do that though.
